hedera-node/hedera-mono-service/src/main/java/com/hedera/services/contracts/execution/HederaMessageCallProcessor.java

```diff
@@ -15,66 +15,42 @@
  */
 package com.hedera.services.contracts.execution;
 
-import static org.hyperledger.besu.evm.frame.ExceptionalHaltReason.INSUFFICIENT_GAS;
-import static org.hyperledger.besu.evm.frame.MessageFrame.State.*;
+import static org.hyperledger.besu.evm.frame.MessageFrame.State.CODE_EXECUTING;
+import static org.hyperledger.besu.evm.frame.MessageFrame.State.EXCEPTIONAL_HALT;
 
 import com.hedera.services.contracts.execution.traceability.ContractActionType;
 import com.hedera.services.contracts.execution.traceability.HederaOperationTracer;
-import com.hedera.services.store.contracts.HederaStackedWorldStateUpdater;
+import com.hedera.services.evm.contracts.execution.HederaEvmMessageCallProcessor;
 import com.hedera.services.store.contracts.precompile.HTSPrecompiledContract;
 import java.nio.charset.StandardCharsets;
-import java.util.HashMap;
 import java.util.Map;
-import java.util.Optional;
 import org.apache.tuweni.bytes.Bytes;
 import org.hyperledger.besu.datatypes.Address;
-import org.hyperledger.besu.datatypes.Wei;
 import org.hyperledger.besu.evm.EVM;
-import org.hyperledger.besu.evm.frame.ExceptionalHaltReason;
 import org.hyperledger.besu.evm.frame.MessageFrame;
 import org.hyperledger.besu.evm.precompile.PrecompileContractRegistry;
 import org.hyperledger.besu.evm.precompile.PrecompiledContract;
-import org.hyperledger.besu.evm.processor.MessageCallProcessor;
 import org.hyperledger.besu.evm.tracing.OperationTracer;
 
-/** Overrides Besu precompiler handling, so we can break model layers in Precompile execution */
-public class HederaMessageCallProcessor extends MessageCallProcessor {
+public class HederaMessageCallProcessor extends HederaEvmMessageCallProcessor {
     private static final String INVALID_TRANSFER_MSG = "Transfer of Value to Hedera Precompile";
-    private static final Optional<ExceptionalHaltReason> ILLEGAL_STATE_CHANGE =
-            Optional.of(ExceptionalHaltReason.ILLEGAL_STATE_CHANGE);
     public static final Bytes INVALID_TRANSFER =
             Bytes.of(INVALID_TRANSFER_MSG.getBytes(StandardCharsets.UTF_8));
 
-    private final Map<Address, PrecompiledContract> hederaPrecompiles;
-
     public HederaMessageCallProcessor(
             final EVM evm,
             final PrecompileContractRegistry precompiles,
             final Map<String, PrecompiledContract> hederaPrecompileList) {
-        super(evm, precompiles);
-        hederaPrecompiles = new HashMap<>();
+        super(evm, precompiles, hederaPrecompileList);
         hederaPrecompileList.forEach((k, v) -> hederaPrecompiles.put(Address.fromHexString(k), v));
```

Comment:
We can remove this row since the super. hederaPrecompiles is already being populated in the HederaEvmMessageCallProcessor's constructor from the same "hederaPrecompileList" object.